### ISSUE TITLE
Missing log labels: instance_id, zone

### ISSUE THREAD

**Issue by b3nk3nobi** [GoogleCloudPlatform/ops-agent#493]
In the Log Explorer there is no information about `instance_id` or `zone`:
![vivaldi__2022_03_24_08-53-05](https://user-images.githubusercontent.com/1058037/159875614-ced387cb-752a-44c6-a5a4-4b4e710519f0.png)

I cannot filter logs by instance:
![vivaldi__2022_03_24_09-38-13](https://user-images.githubusercontent.com/1058037/159876300-680cb695-a575-4656-924b-554f887734a3.png)

VMs don't have any service account assigned and it cannot be changed because they are in production (can't stop them), so I'm using file with SA private key located in `/etc/google/auth/application_default_credentials.json`. 

Using `curl` to get these values on the actual VM is working fine:
![WindowsTerminal__2022_03_24_09-00-20](https://user-images.githubusercontent.com/1058037/159877229-7ae288a4-ac95-4e6b-b916-d596841eb3bb.png)

- `logging-module.log`:
```
[2022/03/24 07:15:34] [ info] [engine] started (pid=192594)
[2022/03/24 07:15:34] [ info] [storage] version=1.1.5, initializing...
[2022/03/24 07:15:34] [ info] [storage] root path '/var/lib/google-cloud-ops-agent/fluent-bit/buffers'
[2022/03/24 07:15:34] [ info] [storage] normal synchronization mode, checksum enabled, max_chunks_up=128
[2022/03/24 07:15:34] [ info] [storage] backlog input plugin: storage_backlog.3
[2022/03/24 07:15:34] [ info] [cmetrics] version=0.2.2
[2022/03/24 07:15:34] [ info] [input:storage_backlog:storage_backlog.3] queue memory limit: 47.7M
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.0] metadata_server set to http://metadata.google.internal
[2022/03/24 07:15:34] [ info] [oauth2] HTTP Status=200
[2022/03/24 07:15:34] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.1] metadata_server set to http://metadata.google.internal
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.0] worker #6 started
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.0] worker #7 started
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.0] worker #5 started
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.0] worker #4 started
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.0] worker #3 started
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.0] worker #2 started
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.0] worker #1 started
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.0] worker #0 started
[2022/03/24 07:15:34] [ info] [oauth2] HTTP Status=200
[2022/03/24 07:15:34] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.1] worker #6 started
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.1] worker #7 started
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.1] worker #5 started
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.1] worker #4 started
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.1] worker #3 started
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.1] worker #2 started
[2022/03/24 07:15:34] [ info] [output:prometheus_exporter:prometheus_exporter.2] listening iface=0.0.0.0 tcp_port=20202
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.1] worker #1 started
[2022/03/24 07:15:34] [ info] [output:stackdriver:stackdriver.1] worker #0 started
[2022/03/24 07:15:34] [ info] [input:tail:tail.1] inotify_fs_add(): inode=2592 watch_fd=1 name=/var/log/syslog
[2022/03/24 07:15:34] [ info] [input:tail:tail.2] inotify_fs_add(): inode=534597 watch_fd=1 name=/var/log/google-cloud-ops-agent/subagents/logging-module.log
[2022/03/24 08:12:16] [ info] [oauth2] HTTP Status=200
[2022/03/24 08:12:16] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
[2022/03/24 08:12:17] [ info] [oauth2] HTTP Status=200
[2022/03/24 08:12:17] [ info] [oauth2] access token from 'www.googleapis.com:443' retrieved
```
- version: `google-cloud-ops-agent 2.12.0~ubuntu20.04`

Am I missing something or is it working as it should be?

**Comment by sophieyfang** [comment]
Have you followed the instructions here: https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/authorization#authorizing_the_ops_agent

**Comment by b3nk3nobi** [comment]
Yes, I have followed instructions. Without it there is a lot of errors about authorization in `logging-module.log`

**Comment by logicbomb421** [comment]
I am also facing this issue. 

I am in a similar situation as @b3nk3nobi where we don't run VMs with attached service accounts, and can't change that. I have followed the instructions in the [authenticating the agent docs](https://cloud.google.com/stackdriver/docs/solutions/agents/ops-agent/authorization#authorizing_the_ops_agent) by creating a service account key, placing it in a secure location, and then setting `GOOGLE_APPLICATION_CREDENTIALS=/path/to/service-account/key.json`. The logging service logs show successful oauth2 communication, and logs _do_ make it to Stackdriver, so I know the key is being picked up.

I am also able to receive the instance ID and zone via the metadata.google.internal HTTP call on the VM in question.

Without the `resource.labels.instance_id` field correctly set, logs don't properly associate with their originating VM, causing issues when creating dashboards, searching logs, etc.

Please let me know if more detail is needed. Thanks!

**Comment by hsmatulisgoogle** [comment]
These are missing due to how fluent bit operates when it is not authenticated through the metadata server, [since it will only auto populate these fields](https://github.com/fluent/fluent-bit/blob/7362d3a03b70304f05c38cea9d37793c092ea9db/plugins/out_stackdriver/stackdriver.c#L1199-L1217) through the `metadata_server_auth` authentication.

If you just want to be able to identify VMs, as a work-around, versions after 2.15.0 include https://github.com/GoogleCloudPlatform/ops-agent/pull/544 which auto populates the resource_name label

Replying to @logicbomb421 's attempted work-around:
* It is true that the exclude_logs and modify_fields processors user the same verifying logic
* That logic will map these fields to the underlying fluent bit labels with the corresponding meaning (these are custom to fluent bit)
  * In this case `logging.googleapis.com/monitored_resource` is the [underlying fluent bit key](https://github.com/fluent/fluent-bit/blob/7362d3a03b70304f05c38cea9d37793c092ea9db/plugins/out_stackdriver/stackdriver.h#L49). Fluent bit will check if a log entry has this key, and if so attempt to use it as a monitored resource spec
  * There is an internal mapping of these that effective behaves as a white list, which does not map the monitored_resource key:
 https://github.com/GoogleCloudPlatform/ops-agent/blob/7cf7426c79cb68f5e50806759c9b010099cde294/confgenerator/filter/internal/ast/ast.go#L51-L59
